File: Sheet/Step 1/Step 1.1/pro1-1-6.py
#answer 1 having time complexity issue
def recur(n):
    if n <= 1:
        return n
    return recur(n - 1) + recur(n - 2)

n = int(input())
print(recur(n))

# #answer 2 using recursion - no time complexity issue

def fibo(n, a={}):
    if n <= 1:
        return n

    if n not in a :
        a[n] = fibo(n - 1, a) + fibo(n - 2, a)

    return a[n]

n = int(input())
print(fibo(n))


# using iteration
def iter(n):
    pre = 0
    cur = 1
    if n==0:
        return n
    elif n==1:
        return n
    else:
        for i in range(1,n):
            prepre = pre
            pre = cur
            cur = prepre + pre

        return cur
        
n = int(input())
z = iter(n) 
print(z)

File: Sheet/Step 1/Step 1.1/c++/pro1-1-6.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{

    int n;
    cin >> n;

    int temp;
    int pre = 0;
    int cur = 1;

    if (n <= 1)
    {
        return n;
    }

    for (int i=2; i <= n; ++i)
    {
        temp = cur;
        cur+=pre;
        pre=temp;
    }
    cout << cur;
    return 0;
}